### PR TITLE
intel_adsp: add missing headers

### PR DESCRIPTION
**Pull request by nashif**
add headers previously included indirectly.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

### CODE REVIEW

**Review by MaureenHelm — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

**Review by microbuilder — DISMISSED** [review]

**Comment by nashif** [comment]
found another one... :(

**Review by henrikbrixandersen — DISMISSED** [review]

**Review by teburd — DISMISSED** [review]

File: drivers/sensor/mchp_tach_xec/tach_mchp_xec.c [hunk 1]
```diff
@@ -8,7 +8,9 @@
 #define DT_DRV_COMPAT microchip_xec_tach
 
 #include <errno.h>
+#include <zephyr/kernel.h>
 #include <zephyr/device.h>
+#include <zephyr/sys/sys_io.h>
```
Comment (gmarull) [inline drivers/sensor/mchp_tach_xec/tach_mchp_xec.c:13]: you're probably looking for `zephyr/arch/cpu.h` instead, this header doesn't expose the `sys_*` APIs (weird, but that's how it is).

Comment (nashif) [reply drivers/sensor/mchp_tach_xec/tach_mchp_xec.c:13]: yes, this is very confusing and I noticed that, we need to fix this.

Comment (nashif) [reply drivers/sensor/mchp_tach_xec/tach_mchp_xec.c:13]: huh, the cpu.h loads arch.h which then includes <zephyr/arch/common/sys_io.h>

**Review by galak — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Pull request merged**